cdk-lib/viewer-stacks/viewer-vpc-stack.ts

```diff
@@ -0,0 +1,85 @@
+import { Construct } from 'constructs';
+import { RemovalPolicy } from 'aws-cdk-lib';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import * as logs from 'aws-cdk-lib/aws-logs';
+import { Stack, StackProps } from 'aws-cdk-lib';
+
+import * as types from '../core/context-types';
+
+export interface ViewerVpcStackProps extends StackProps {
+    readonly captureTgw: ec2.CfnTransitGateway;
+    readonly captureVpc: ec2.Vpc;
+    readonly viewerVpcPlan: types.VpcPlan;
+}
+
+export class ViewerVpcStack extends Stack {
+  public readonly vpc: ec2.Vpc;
+  public readonly flowLog: ec2.FlowLog;
+
+  constructor(scope: Construct, id: string, props: ViewerVpcStackProps) {
+    super(scope, id, props);
+
+    // The VPC the Viewer Nodes will live in
+    this.vpc = new ec2.Vpc(this, 'VPC', {
+        ipAddresses: ec2.IpAddresses.cidr(props.viewerVpcPlan.cidr.block),
+        maxAzs: props.viewerVpcPlan.numAzs,
+        subnetConfiguration: [
+            {
+                subnetType: ec2.SubnetType.PUBLIC,
+                name: 'Ingress',
+                cidrMask: props.viewerVpcPlan.publicSubnetMask
+            },
+            {
+                subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS,
+                name: 'ViewerNodes'
+            }
+        ]
+    });
+
+    // Connects the Viewer VPC to the Capture VPC via its TGW
+    const tgwAttachment = new ec2.CfnTransitGatewayAttachment(this, 'TGWAttachment', {
```

Comment:
Viewer-TGWAttachment? Does it matter? If we have multiple clusters sharing a TGW in the future should it include cluster name?

Reply:
Same comment - the name should have its stack context prepended.